test/lib/mayaUsd/render/mayaToHydra/testPolygonPrimitives.py

```diff
@@ -0,0 +1,361 @@
+# Copyright 2024 Autodesk
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import maya.cmds as cmds
+import fixturesUtils
+import mtohUtils
+
+import platform
+
+class TestPolygonPrimitives(mtohUtils.MtohTestCase):
+    # MayaHydraBaseTestCase.setUpClass requirement.
+    _file = __file__
+
+    @property
+    def imageDiffFailThreshold(self):
+        return 0.1
+    
+    @property
+    def imageDiffFailPercent(self):
+        # HYDRA-837 : Wireframes seem to have a slightly different color on macOS. We'll increase the thresholds
+        # for that platform specifically for now, so we can still catch issues on other platforms.
+        if platform.system() == "Darwin":
+            return 2
+        return 0.5
+
+    def compareSnapshot(self, referenceFilename, cameraDistance=15):
+        self.setBasicCam(cameraDistance)
+        cmds.refresh()
+        self.assertSnapshotClose(referenceFilename, self.imageDiffFailThreshold, self.imageDiffFailPercent)
+
+    def setupScene(self, polygonCreationCallable):
+        self.setHdStormRenderer()
+        polyCreatorNodeName = polygonCreationCallable()[1]
```

Comment:
Now only returns the polygon primitive creation node name, so we can remove the [1] indexing in each test.